src/components/Comment/AddComment.vue

```diff
@@ -65,16 +79,29 @@ export default {
   },
   methods: {
     save() {
-      if (!this.rating) return;
+      if (this.showRating) {
+        if (!this.rating) return;
+      }
 
-      this.onSave({rating: this.rating, comment: this.comment, user: this.signedInUser.userName})
-      .then(() => {
-        this.$emit('saved');
-        this.reset();
-        eventBus.$emit('show-toast', {body: messages.comment.commentAddSuccess, title: messages.generic.success});
-      }).catch(e => {
-        eventBus.$emit('show-toast', {body: e.message, title: messages.generic.error, type: ToastType.ERROR});
-      });
+      if (this.showRating) {
+        var payload = {
+          rating: this.rating,
+          comment: this.comment,
+          user: this.signedInUser.userName
+        }
+      } else {
+        var payload = {
+          toolId: this.toolId,
+          comment: this.$refs.commentBox.editedValue,
+          username: this.signedInUser.username
+        }
+      }
+      if(this.isEdit) {
+        this.onSave(payload, this.isEdit, this.commentId, this.commentIndex);
+      } else {
+        this.onSave(payload, this.isEdit, this.commentId, this.commentIndex);
```

Comment:
both the if and else conditions look same. What should be different in these blocks?